miden-lib/src/transaction/inputs.rs

```diff
@@ -295,29 +301,33 @@ fn add_input_notes_to_advice_inputs(
 
         inputs.extend_map([(assets.commitment(), assets.to_padded_assets())]);
 
-        // NOTE: keep in sync with the `prologue::process_input_node` kernel procedure
+        // NOTE: keep in sync with the `prologue::process_input_note_details` kernel procedure
         note_data.extend(recipient.serial_num());
         note_data.extend(*recipient.script().hash());
         note_data.extend(*recipient.inputs().commitment());
         note_data.extend(*assets.commitment());
 
-        note_data.extend(Word::from(note.metadata()));
+        // NOTE: keep in sync with the `prologue::process_note_args_and_metadata` kernel procedure
         note_data.extend(Word::from(*note_arg));
+        note_data.extend(Word::from(note.metadata()));
 
+        // NOTE: keep in sync with the `prologue::process_note_assets` kernel procedure
         note_data.push((assets.num_assets() as u32).into());
         note_data.extend(assets.to_padded_assets());
 
         // insert note authentication path nodes into the Merkle store
         match input_note {
             InputNote::Authenticated { note, proof } => {
+                // NOTE: keep in sync with the `prologue::process_input_note` kernel procedure
+                note_data.push(Felt::ONE);
```

Comment:
I would also mention that this is the `is_authenticated` flag.

Reply:
added